department-of-veterans-affairs/caseflow
# Enable bulk/single assign from queue for CAMO

Resolves [CASEFLOW-2249](https://vajira.max.gov/browse/CASEFLOW-2249), [CASEFLOW-2284](https://vajira.max.gov/browse/CASEFLOW-2284), and [CASEFLOW-2285](https://vajira.max.gov/browse/CASEFLOW-2285)

### Description
Assign page functionality for CAMO team members. Allows for single and bulk task assignment from a single view without having to visit case details pages individually. Functionality behind the `vha_predocket_workflow` feature toggle.

### Acceptance Criteria
- [ ] CAMO is able to assign multiple cases to a program office from the assign page
- [ ] CAMO is able to assign a single case to a program office from the assign page

### Testing Plan
1. Intake a handful (4+) of appeals w/ VHA issues
2. Visit the CAMO org queue (`/organizations/vha-camo`)
3. Use the "Switch Views" dropdown and select the "Assign VHA CAMO team cases" option
4. Ensure you're redirected to `/queue/CAMOUSER/assign?role=camo`
5. Select 2-3 cases to assign to a Program office
6. Ensure the "Assign" button updates with the correct amount of selected appeal
7. Ensure all of the program offices populate in the "Select a program office" dropdown
8. Choose a program office to assign to and click "Assign"
9. Ensure a success message displays with the correct number of tasks and the correct program office
10. Repeat for a single case
11. Ensure the cases/tasks went to the appropriate program offices

### User Facing Changes
<img width="1772" alt="Screen Shot 2021-11-17 at 10 54 56 AM" src="https://user-images.githubusercontent.com/640182/142271689-6a152ab5-a353-4c22-8b4a-fb23686dbe3b.png">
<img width="1771" alt="Screen Shot 2021-11-17 at 10 55 34 AM" src="https://user-images.githubusercontent.com/640182/142271716-4c01869d-e51b-42ee-95a0-a267d8620662.png">
<img width="1770" alt="Screen Shot 2021-11-17 at 10 56 26 AM" src="https://user-images.githubusercontent.com/640182/142271729-b45b8aae-c397-4586-9129-dc75c037c500.png">
<img width="1767" alt="Screen Shot 2021-11-17 at 10 56 37 AM" src="https://user-images.githubusercontent.com/640182/142271738-ed1ad45b-912a-40ff-9f0f-825003baac99.png">
